### PR TITLE
refactor(dart_frog_cli): introduce dev server runner

### PR DESCRIPTION
**Pull request by renancaraujo**
## Status

**READY**

## Description

This PR extracts the dev server logic from the DevCommand class to a standalone runner class. 

This is part of #705

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/dart_frog_cli/lib/src/dev_server_runner/restorable_directory_generator_target.dart [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import 'dart:collection';
```
Comment (renancaraujo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/restorable_directory_generator_target.dart:1]: The code in this file is identical to a subsection of what we had on `packages/dart_frog_cli/lib/src/commands/dev/dev.dart`

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            _dartVmServiceAlreadyInUseErrorRegex.hasMatch(message);
+        final isSDKWarning = _warningRegex.hasMatch(message);
+
+        if (isDartVMServiceAlreadyInUseError) {
+          logger.err(
+            '$message '
+            '''Try specifying a different port using the `--dart-vm-service-port` argument when running `dart_frog dev`.''',
+          );
+        } else if (isSDKWarning) {
+          /// Do not kill the process if the error is a warning from the SDK.
```
Comment (erickzanardo) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:191]:
```suggestion
          // Do not kill the process if the error is a warning from the SDK.
```

**Review by erickzanardo — DISMISSED** [review]

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart (hunk 2 above)
Comment (renancaraujo) [reply packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:191]: lol, in my defense this was like that ion the dev command

**Review by alestiago — DISMISSED** [review]
LGTM as long as the tests have been preserved from the previous `dev_test.dart` content and they have been solely moved around.

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    } else {
+      logger.detail('[process] process.kill()...');
+      process.kill();
+    }
+    logger.detail('[process] killing process complete.');
+  }
+
+  // TODO(renancaraujo): this method returns a future that completes when the
+  // process is killed, but it should return a future that completes when the
+  // process is finished starting.
```
[the remark this answers is not in the document]
Comment (renancaraujo) [reply packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:149]: No need, tracking daemon refactors on #705

[the remark this answers is not in the document]
Comment (renancaraujo) [reply packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:148]: CHANGE ADDED ON https://github.com/VeryGoodOpenSource/dart_frog/pull/730

**Review by erickzanardo — DISMISSED** [review]

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**